ivy/functional/backends/paddle/experimental/statistical.py

```diff
@@ -58,9 +44,12 @@ def median(
             axis = None
     if (input.ndim == 1 or axis is None) and not keepdims:
         ret = ret.squeeze()
-    return ret.astype(ret_dtype)
+    return ret.astype(input.dtype)
 
 
+@with_supported_dtypes(
+    {"2.5.1 and below": ("complex", "float32", "float64", "int64")}, backend_version
+)
 def nanmean(
```

Comment:
when I run the `test_nanmean` it fails because the ground truth backend convert all returned dtypes to `float32` or `float64` but the other backend return the same input dtype normally, so what should I do here ? do I cast the return types in `test_nanmean` to make it pass ?

Reply:
I'd say it would be good to fix the ground truth backend (probably in a different PR)